src/main/java/com/gorlah/kappabot/function/slashdot/MobileSlashdotTitleFunction.java

```diff
@@ -18,14 +18,15 @@
 
     @Override
     public Optional<BotResponse> process(BotRequestMetadata metadata) {
-        if (slashdotParser.containsMobileSlashdotUrl(metadata.getMessage())) {
-            var response = slashdotParser.getTitles(metadata.getMessage()).stream()
-                    .map(title -> "> " + title)
-                    .collect(Collectors.joining("\n"));
-            if (!response.trim().isEmpty()) {
-                return Optional.of(BotResponses.fromMarkdown(response));
-            }
-        }
-        return Optional.empty();
+        return Optional.of(metadata)
+                .map(BotRequestMetadata::getMessage)
+                .map(slashdotParser::getTitles)
+                .filter(titles -> !titles.isEmpty())
+                .map(titles -> titles.stream()
+                        .map(title -> "> " + title)
```

Comment:
What new markdown format do you mean? The The `>` is a markdown thing in general for block quotes it is not specific to Discord.

Reply:
What I meant was, if this was a response to an IRC message, would you want the `>` to be there? Isn't that what your last pr was about?

Reply:
That will get handled in the `MarkdownToPlainTextResponseFormatter`. I set it up so a `BotFunction` can either respond with a message in a specific desired format directly (it could optionally directly look at where the message came from and respond differently), **or** provide a response in a general markup like Markdown and have it converted automatically depending on the expected content type declared by the requester (this happens in the `ResponseFormatter`).

So if the request comes from an IRC source, it would get back a message without the `>` and if the request came from Discord, it would leave the `>` in. This is the flexibility I was going for with my previous rounds of changes.

Reply:
Okay, that's awesome. Good job on that!